app/models/manageiq/providers/redfish/physical_infra_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,38 @@
+require "server_sent_events"
+
+module ManageIQ::Providers::Redfish
+  class PhysicalInfraManager::EventCatcher::Runner \
+      < ManageIQ::Providers::BaseManager::EventCatcher::Runner
+    def monitor_events
+      event_stream.listen do |event|
+        event_monitor_running
+        @queue << event
+      end
+    end
+
+    def stop_event_monitor
+    end
+
+    def queue_event(events)
+      PhysicalInfraManager::EventParser.events_to_hashes(
+        events, @cfg[:ems_id]
```

Comment:
Isn't a single event queued not a set?  Shouldn't this be `hash = ManageIQ::Providers::Redfish::PhysicalInfraManager::EventParser.event_to_hash(event, @cfg[:ems_id])` then `EmsEvent.add_queue("add", @cfg[:ems_id], hash)`

Reply:
This is a direct consequence of Redfish API being a bit on stupid on occasions (and this particular stupidity is documented [here][]). Event in Redfish jargon is actually a set of event records and this is what I get as an event in `monitor_events`.

   [here]: http://redfish.dmtf.org/schemas/DSP0266_1.5.0.html#eventservice

To handle this, I either need to parse the json data in `monitor_events` (which just feels wrong) or do it here (which still feels wrong, but a bit less so).

Thinking about this now, I should probably add another layer of indirection here and bury those ugly details in redfish client as you already suggested in some other comment. Which is exactly what I will do.